src/snowflake/cli/plugins/nativeapp/teardown_processor.py

```diff
@@ -89,9 +113,38 @@ def drop_application(self, auto_yes: bool):
                     cc.message(f"Did not drop application object {self.app_name}.")
                     return  # The user desires to keep the app, therefore exit gracefully
 
-        # 4. All validations have passed, drop object
+        # 4. Check for application objects owned by the application
+        application_objects = self.get_objects_owned_by_application()
+        if len(application_objects) > 0:
+            application_objects_str = self._application_objects_to_str(
+                application_objects
+            )
+            if cascade is True:
+                cc.message(
+                    f"The following objects are owned by application {self.app_name} and will be dropped:\n{application_objects_str}"
+                )
+            elif cascade is False:
+                cc.message(
+                    f"The following objects are owned by application {self.app_name} but will not be dropped:\n{application_objects_str}"
+                )
```

Comment:
As discussed, is this ever true? Will `drop application` allow orphaned objects?

Reply:
`drop application` will not allow orphaned objects, we expect this command to fail.
Changed the wording to just reflect the state in case the behavior will change in the future (automatic ownership transfer, for example)